ob1k-core/src/main/java/com/outbrain/ob1k/common/marshalling/MessagePackRequestMarshaller.java

```diff
@@ -263,14 +263,14 @@ public HttpContent marshallResponsePart(final Object res, final HttpResponseStat
       final Template<T> template = (Template<T>) msgPack.lookup(type);
       return template.read(new Converter(msgPack, value), null);
 
-    } else if (Arrays.equals(ChunkHeader.ERROR_HEADER.getBytes(), header)) {
+    } else if (Arrays.equals(ChunkHeader.ERROR_HEADER.getBytes(CharsetUtil.UTF_8), header)) {
 
       final byte[] body = new byte[remaining];
       byteBufferBody.get(body);
-      throw new RuntimeException(new String(body));
+      throw new RuntimeException(new String(body,CharsetUtil.UTF_8));
```

Comment:
Yes but why ? There are a few changes needed to be made, some places have the encoding and we do need to think about it as it is an open source